### ISSUE TITLE
Abnormal task status

### ISSUE THREAD

**Issue by Ocean456** [daleao/modular-overhaul#177]
![image](https://github.com/daleao/modular-overhaul/assets/95454779/4d01df7e-7065-4dfd-8eb6-3f885bde36b8)

Task stuck, unable to end task

![image](https://github.com/daleao/modular-overhaul/assets/95454779/5f1ebd21-a8ac-4880-8eda-0bbc94ce4cbb)

This is the log: https://smapi.io/log/f08e15a87bfa4c5e838cb556dce70db3

**Comment by Ocean456** [comment]
When I reached all the virtue conditions, I went to the mine to get the Black Sword, and the next day after triggering the mission, it turned into this.

**Comment by daleao** [comment]
I don't understand. The sword is obtained before the quest. So how could you complete the virtue conditions before going to get the sword?

**Comment by Ocean456** [comment]
> I don't understand. The sword is obtained before the quest. So how could you complete the virtue conditions before going to get the sword?

Sorry, what I mean is that before I obtained the Dark Sword, I had met all the conditions required for the hero mission, the five virtue conditions. Then I went to the mine to get the Dark Sword, and the next day Wizard sent me an email. After that, I interacted with the Yoba Altar and successfully purified the Dark Sword. But the task status has changed to this and there is no way to end it
